modules/loader/src/test/java/com/opengamma/strata/loader/csv/TradeCsvLoaderTest.java

```diff
@@ -1255,6 +1256,219 @@ public void test_load_swap_defaultFixedLegDayCount() {
+        .put("Leg 2 Start Date", "2017-05-02")
+        .put("Leg 2 End Date", "2022-05-02")
+        .put("Leg 2 Date Convention", "Following")
+        .put("Leg 2 Date Calendar", "GBLO")
+        .put("Leg 2 Frequency", "6M")
+        .put("Leg 2 Currency", "GBP")
+        .put("Leg 2 Notional", "1000000")
+        .put("Leg 2 Index", "CHF-LIBOR-6M")
+        .build();
+    String csv = Joiner.on(',').join(csvMap.keySet()) + "\n" + Joiner.on(',').join(csvMap.values());
+
+    TradeCsvLoader test = TradeCsvLoader.standard();
+    ValueWithFailures<List<SwapTrade>> result = test.parse(ImmutableList.of(CharSource.wrap(csv)), SwapTrade.class);
+    assertEquals(result.getFailures().size(), 1, result.getFailures().toString());
+    FailureItem failure = result.getFailures().get(0);
+    assertEquals(failure.getReason(), FailureReason.PARSING);
+    assertEquals(
+        failure.getMessage(),
+        "CSV file trade could not be parsed at line 2: Swap leg must not define both 'Leg 1 Fixed Rate' or 'Leg 1 Known Amount' and  'Leg 1 Index'");
+  }
+
+  public void test_load_swap_fixedRateAndIndex() {
+    ImmutableMap<String, String> csvMap = ImmutableMap.<String, String>builder()
+        .put("Strata Trade Type", "Swap")
+        .put("Id Scheme", "OG")
+        .put("Id", "1234")
+        .put("Trade Date", "20170101")
+        .put("Trade Time", "12:30")
+        .put("Trade Zone", "Europe/Paris")
+
+        .put("Leg 1 Direction", "Pay")
+        .put("Leg 1 Start Date", "2017-05-02")
+        .put("Leg 1 End Date", "2022-05-02")
+        .put("Leg 1 Date Convention", "Following")
+        .put("Leg 1 Date Calendar", "GBLO")
+        .put("Leg 1 Frequency", "12M")
+        .put("Leg 1 Currency", "GBP")
+        .put("Leg 1 Notional", "1000000")
+        .put("Leg 1 Fixed Rate", "1.1")
+        .put("Leg 1 Index", "GBP-LIBOR-6M")
+
+        .put("Leg 2 Direction", "Pay")
+        .put("Leg 2 Start Date", "2017-05-02")
+        .put("Leg 2 End Date", "2022-05-02")
+        .put("Leg 2 Date Convention", "Following")
+        .put("Leg 2 Date Calendar", "GBLO")
+        .put("Leg 2 Frequency", "6M")
+        .put("Leg 2 Currency", "GBP")
+        .put("Leg 2 Notional", "1000000")
+        .put("Leg 2 Index", "CHF-LIBOR-6M")
+        .build();
+    String csv = Joiner.on(',').join(csvMap.keySet()) + "\n" + Joiner.on(',').join(csvMap.values());
+
+    TradeCsvLoader test = TradeCsvLoader.standard();
+    ValueWithFailures<List<SwapTrade>> result = test.parse(ImmutableList.of(CharSource.wrap(csv)), SwapTrade.class);
+    assertEquals(result.getFailures().size(), 1, result.getFailures().toString());
+    FailureItem failure = result.getFailures().get(0);
+    assertEquals(failure.getReason(), FailureReason.PARSING);
+    assertEquals(
+        failure.getMessage(),
+        "CSV file trade could not be parsed at line 2: Swap leg must not define both 'Leg 1 Fixed Rate' or 'Leg 1 Known Amount' and  'Leg 1 Index'");
+  }
+
```

Comment:
It would be worth having a test to show that variable known amount works. The code for variable known amount already exists AFAICT, but I assume it is untested.